### PR TITLE
Tidy backend section

### PR DESCRIPTION
**Pull request by Arwel**
Brief tidyup of the backend section. Still unsure as to if the content is levelled right, and if it needs more detail.

### CODE REVIEW

**Review by Paul2e — APPROVED** [review]

File: frameworks/engineering/backend.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ topics:
     content:
       - level: 1
         criteria:
-          - "Provides regular status updates to their mentor/buddy"
+          - "Provides regular updates to their mentor/buddy"
           - "Points out syntactical improvements in code reviews"
           - "Writes PR descriptions that provide basic context for the change"
           - "Seeks guidance from other engineers, rather than answers"
```
Comment (AlexMaher) [inline frameworks/engineering/backend.md:18]: Is it worth being specific and saying we expect them to have investigated some options to present for guidance?  Or am I being too picky?

File: frameworks/engineering/backend.md [hunk 2]
```diff
@@ -24,8 +24,7 @@ topics:
           - "Accepts feedback graciously"
           - "Gives feedback to peers when asked"
           - "Provides helpful and actionable feedback in code reviews in an empathetic manner"
-          - "Writes PR descriptions that provide context and provide rationale for significant decisions"
-          - "Can deliver their work to their team and others"
+          - "Writes PR descriptions that provide context and rationale for significant decisions"
           - "Proactively gives feedback to those they work with"
           - "Brave enough to ask questions, and appear ignorant."
```
Comment (AlexMaher) [inline frameworks/engineering/backend.md:29]: Dont think it needs the ignorant bit.

File: frameworks/engineering/backend.md [hunk 3]
```diff
@@ -121,7 +121,7 @@ topics:
           - "Asks why. Does not take truths for granted unless they understand exactly where they are coming from (especially with regards to regulation, compliance, etc)"
           - "Drives changes to engineering practices with well-reasoned arguments and a 'strong opinion, weakly held' mentality"
```
Comment (AlexMaher) [inline frameworks/engineering/backend.md:122]: Potentially evidence based?  I like the 'strong opinion, weakly held' bit

File: frameworks/engineering/backend.md [hunk 4]
```diff
@@ -153,6 +151,7 @@ topics:
           - "Considers metrics when building a new system"
```
Comment (AlexMaher) [inline frameworks/engineering/backend.md:151]: Are these success of performance, the former should probably be in an earlier level?

**Pull request merged**